app/javascript/components/upload/WizardNavPanel.jsx

```diff
@@ -176,6 +170,52 @@ function clusteringHelpContent(isClusteringParsing) {
   </Popover>
 }
 
+/**
+ * Return the appropriate display of the main steps based on traditional or AnnData upload experience
+ */
+function mainStepsDisplay(formState, serverState, currentStep, setCurrentStep, mainSteps,
```

Comment:
Trivial blocker: React component names start with a capital letter per SCP convention.
```suggestion
function MainStepsDisplay(formState, serverState, currentStep, setCurrentStep, mainSteps,
```
```suggestion
function mainStepsDisplay(formState, serverState, currentStep, setCurrentStep, mainSteps,
```